File: 백준/Bronze/2798. 블랙잭/블랙잭.cc
#include <iostream>
#include <algorithm>
using namespace std;

int a[102];
int c[102];

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n, m;
    cin>>n>>m;
    for(int i=0; i<n; i++) cin>>a[i];
    sort(a,a+n);

    int ans=0;
    for(int i=0; i<n-2; i++){
        for(int j=i+1; j<n-1; j++){
            for(int k=j+1; k<n; k++){
                //cout<<i<<j<<k;
                int sum = a[i]+a[j]+a[k];
                if(sum>m) continue; //m을 넘으면 안됨
                ans = max(ans, sum);
            }
        }
    }

    cout<<ans;
}

File: 백준/Bronze/2798. 블랙잭/블랙잭.java
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	private Scanner sc = new Scanner(System.in);
	
	private void solve() throws FileNotFoundException {
	    //Scanner sc = new Scanner(new FileInputStream("input.txt"));
	    
	    int n = sc.nextInt();
	    int m = sc.nextInt();
	    
	    int[] arr = new int[n];
	    for(int i=0; i<n; i++) {
	    	arr[i] = sc.nextInt();
	    }
	    
	    int ans =0;
	    for(int i=0; i<n-2; i++) {
	    	for(int j=i+1; j<n-1; j++) {
	    		for(int k=j+1; k<n; k++) {
	    			int sum = arr[i]+arr[j]+arr[k];
	    			if(sum<=m) ans= Math.max(ans, sum);
	    		}
	    	}
	    }
	    System.out.println(ans);
	    
	    sc.close();
	}

	public static void main(String[] args) throws Exception {
	    new Main().solve();
	}
}